File: backend/src/main/java/com/personal/backend/entity/OperationLog.java
package com.personal.backend.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 操作日志表：记录用户关键操作，用于审计追溯
 */
@Data
@TableName("operation_log")
public class OperationLog {

    @TableId(type = IdType.AUTO)
    private Long id;

    /** 操作者 */
    private Long userId;

    /** 操作者类型：1=业务用户 user 表 / 2=开发账号 admin_user 表 */
    private Integer userType;

    /** 操作模块：EXPENSE/WEIGHT/LEARN/USER/NOTE/ADMIN */
    private String module;

    /** 操作动作：CREATE/UPDATE/DELETE/RESTORE/LOGIN/REGISTER */
    private String action;

    /** 操作对象 ID */
    private Long targetId;

    /** 操作内容描述 */
    private String content;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    @TableLogic
    private Integer deleted;

    /** 操作人显示名（非表字段，查询时由 Service 填充：业务用户昵称 / 开发账号昵称） */
    @TableField(exist = false)
    private String operatorName;
}

File: front/src/api/operationLog.ts
/**
 * 操作日志接口
 */
import { requestApi } from './request'
import type { PageResult } from './types'

/** 操作日志记录 */
export interface OperationLog {
  id: number
  userId: number
  /** 操作模块：EXPENSE/WEIGHT/LEARN/USER/NOTE/EXERCISE */
  module: string
  /** 操作动作：CREATE/UPDATE/DELETE/RESTORE/LOGIN/REGISTER */
  action: string
  targetId: number | null
  content: string
  createdAt: string
  updatedAt: string
}

/** 分页查询操作日志（silent=true 时静默，用于导航栏取总数等后台加载） */
export function listOperationLogs(params: {
  module?: string
  action?: string
  startDate?: string
  endDate?: string
  page?: number
  size?: number
  silent?: boolean
}) {
  const { silent, ...rest } = params
  return requestApi<PageResult<OperationLog>>({
    url: '/operation-logs',
    method: 'GET',
    params: rest,
    silent
  })
}

/** 操作日志业务字典 */
export const LOG_MODULES = ['EXPENSE', 'WEIGHT', 'LEARN', 'USER', 'NOTE', 'EXERCISE', 'FOOD'] as const
export const LOG_ACTIONS = ['CREATE', 'UPDATE', 'DELETE', 'RESTORE', 'LOGIN', 'REGISTER'] as const

/** 模块中文显示 */
export const MODULE_LABELS: Record<string, string> = {
  EXPENSE: '记账',
  WEIGHT: '健康',
  LEARN: '学习',
  USER: '用户',
  NOTE: '笔记',
  EXERCISE: '锻炼',
  FOOD: '饮食'
}

/** 动作中文显示 */
export const ACTION_LABELS: Record<string, string> = {
  CREATE: '新增',
  UPDATE: '修改',
  DELETE: '删除',
  RESTORE: '恢复',
  LOGIN: '登录',
  REGISTER: '注册'
}